packages/cli/src/standalone-impl.ts

```diff
@@ -13,32 +15,90 @@ import {
 } from '@pipcook/pipcook-core';
 import * as fs from 'fs-extra';
 import * as path from 'path';
-export class DataAccessorImpl<T> implements DataAccessor<T> {
+export class DataAccessorImpl<T> implements RandomDataAccessor<T> {
+  dataLength: number;
+  randomIndex: number;
+  randomMap: Array<number>;
   constructor(
     private dataAccessor: DataAccessor<T>
   ) {}
+  init(size: number): void {
+    this.dataLength = size;
+    this.randomIndex = 0;
+    this.randomMap = this.createRandom(this.dataLength);
+  }
   next(): Promise<Sample<T> | null> {
     return this.dataAccessor.next();
   }
   nextBatch(batchSize: number): Promise<Array<Sample<T>> | null> {
     return this.dataAccessor.nextBatch(batchSize);
   }
+  nextRandom(): Promise<Sample<T> | null> {
+    if ( this.randomIndex >= this.dataLength ) {
+      this.dataAccessor.seek(this.randomIndex++);
+    } else {
+      this.dataAccessor.seek(this.randomMap[this.randomIndex++]);
+    }
+    return this.dataAccessor.next();
+  }
+  nextBatchRandom(batchSize: number): Promise<Array<Sample<T>> | null> {
+    const buffer = [];
+    while (batchSize) {
+      buffer.push(this.nextRandom());
+      batchSize--;
+    }
+    return Promise.all(buffer);
+  }
   seek(pos: number): Promise<void> {
     return this.dataAccessor.seek(pos);
   }
+  createRandom(dataLength = 0): Array<number> {
+    if (dataLength === 0)
+      return [];
+    let i = 0;
+    let index = 0;
+    let temp: any | never = null;
+    let arr: Array<number | never> = [];
+    for ( i = 0; i < dataLength; i++ ) {
+      arr[ i ] = i;
+    }
+    for ( i = 0; i < dataLength; i++ ) {
+      index = Math.floor( Math.random() * (dataLength - i) ) + i;
```

Comment:
Should use `seedrandom.js` instead of `Math.random()` for a more controllable random behavior.